geode-core/src/test/java/org/apache/geode/internal/cache/entries/AbstractRegionEntryTest.java

```diff
@@ -131,25 +137,30 @@ public void gatewayEventsFromSameDSShouldThrowCMEIfMisordered() {
     EntryEventImpl entryEvent1 = new EntryEventImpl();
     entryEvent1.setRegion(lr);
     when(lr.getCache()).thenReturn(cache);
-    when(cache.getGatewayConflictResolver()).thenReturn(null);
+    GatewayConflictResolver resolver = mock(GatewayConflictResolver.class);
+    when(cache.getGatewayConflictResolver()).thenReturn(resolver);
 
     VersionTag tag1 = VersionTag.create(member1);
     tag1.setVersionTimeStamp(1);
-    tag1.setDistributedSystemId(2);
+    tag1.setDistributedSystemId(3);
     tag1.setIsGatewayTag(true);
 
     VersionTag tag2 = VersionTag.create(member1);
     tag2.setVersionTimeStamp(2);
-    tag2.setDistributedSystemId(2);
+    tag2.setDistributedSystemId(3);
     tag2.setIsGatewayTag(true);
 
     ((TestableRegionEntry) re).setVersions(tag2);
     assertEquals(tag2.getVersionTimeStamp(),
         re.getVersionStamp().asVersionTag().getVersionTimeStamp());
+    assertEquals(3, ((TestableRegionEntry) re).getDistributedSystemId());
 
     // apply tag1 with smaller timestamp should throw CME
     entryEvent1.setVersionTag(tag1);
+    expectedException.expect(ConcurrentCacheModificationException.class);
+    expectedException.expectMessage("conflicting WAN event detected");
     re.processVersionTag(entryEvent1);
+    verify(resolver, never()).onEvent(any(), any());
```

Comment:
Would you be able to separate this into a separate test?  It will make the test more granular and easier to debug